core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala

```diff
@@ -83,6 +83,8 @@ class CheckpointFile[T](val file: File, version: Int, formatter: CheckpointFileF
         if (line == null)
           return Seq.empty
         line.toInt match {
+          // Backticks are required to match against the value of the version field instead of introducing a new
+          // variable called version
```

Comment:
Oh, in that case, would it be clearer to just rename version in the constructor to sth like expectedVersion and then match to version here?